fofix/game/Song.py

```diff
@@ -2104,23 +2108,31 @@ def __init__(self, engine, infoFileName, songTrackName, guitarTrackName, rhythmT
         except Exception, e:
             Log.warn("Unable to load rhythm track: %s" % e)
 
-
+        i = 3
         try:
-            if drumTrackName:
-                self.drumTrack = Audio.StreamingSound(self.engine.audio.getChannel(3), drumTrackName)
+            if drumTrackNames:
+                for drumTrackName in drumTrackNames:
```

Comment:
I tried what you have proposed.  It actually cost me an extra line of code
because I have to maintain the index outside of the try.

 i = 3
        try:
            if drumTrackNames:
                for j, drumTrackName in enumerate(drumTrackNames):
                    i = j + 3

self.drumTracks.append(Audio.StreamingSound(self.engine.audio.getChannel(i),
drumTrackName))
                    i += 1

On Sat, Mar 7, 2015 at 3:18 PM, Matthew Sitton <notifications@github.com>
wrote:

> In fofix/game/Song.py
> <https://github.com/fofix/fofix/pull/33#discussion_r25999696>:
>
> >          try:
> > -            if drumTrackName:
> > -                self.drumTrack = Audio.StreamingSound(self.engine.audio.getChannel(3), drumTrackName)
> > +            if drumTrackNames:
> > +                for drumTrackName in drumTrackNames:
>
> In this loop instead of what you have you could use enumerate()
>
> for i, drumTrackName in enumerate(drumTrackNames):
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/fofix/fofix/pull/33/files#r25999696>.
>